### PR TITLE
Add mock http calls to DiscordValidatorTests

### PR DESCRIPTION
**Pull request by Bpendragon**
Tests all return paths of the call.

### CODE REVIEW

File: Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs [hunk 1]
```diff
@@ -30,5 +35,68 @@ public void DiscordApiCredentialsValidator_Test()
...
+                if (!message.Equals(testCase.ExpectedMessage))
+                {
+                    sb.AppendLine($"The test case '{testCase.Title}' was expecting '{testCase.ExpectedMessage}' but found '{message}'.");
+                }
+            }
+
+            sb.Length.Should().Be(0, sb.ToString());
+        }
+
     }
```
Comment (eddynaka) [inline Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs:101]: you can remove this new line

**Comment by eddynaka** [comment]
>                         return ReturnUnexpectedResponseCode(ref message, response.StatusCode);

you can do something like:
```
ReturnUnexpectedResponseCode(ref message, response.StatusCode, account: id);
```

---
Refers to: Src/Plugins/Security/SEC101_046.DiscordApiCredentialsValidator.cs:115 in 484a0e8. [](commit_id = 484a0e8f3644ef53600489e2b420e068b21068ef, deletion_comment = False)

**Comment by eddynaka** [comment]
>                 return ReturnUnhandledException(ref message, e);

same here:
```
account: id
```

---
Refers to: Src/Plugins/Security/SEC101_046.DiscordApiCredentialsValidator.cs:121 in 484a0e8. [](commit_id = 484a0e8f3644ef53600489e2b420e068b21068ef, deletion_comment = False)

File: Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs (hunk 1 above)
Comment (Bpendragon) [reply Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs:101]: Fixed in next iteration

**Comment by Bpendragon** [comment]
>                         return ReturnUnexpectedResponseCode(ref message, response.StatusCode);

Closed per discussion offline with eddy. ReturnUnexpectedResponseCode may need refectored for output and input to align better. 

---
In reply to: [891391044](https://github.com/microsoft/sarif-pattern-matcher/pull/523#issuecomment-891391044) [](http://example.com/codeflow?ancestors=891391044,891391044)

---
Refers to: Src/Plugins/Security/SEC101_046.DiscordApiCredentialsValidator.cs:115 in 484a0e8. [](commit_id = 484a0e8f3644ef53600489e2b420e068b21068ef, deletion_comment = False)

File: Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs [hunk 2]
```diff
@@ -30,5 +35,67 @@ public void DiscordApiCredentialsValidator_Test()
...
+
+            var sb = new StringBuilder();
+            foreach (var testCase in testCases)
+            {
+                string message = string.Empty;
+                ResultLevelKind resultLevelKind = default;
+                var fingerprint = new Fingerprint(fingerprintText);
+                var keyValuePairs = new Dictionary<string, string>();
+
+                MockHelper.ResetStaticInstance<SquareCredentialsValidator>();
```
Comment (eddynaka) [inline Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs:79]: >SquareCredentialsValidator [](http://example.com/codeflow?start=47&length=26)

this should be DiscordApiCredentialsValidator

**Review by eddynaka — CHANGES_REQUESTED** [review]
:clock1:

File: Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs (hunk 2 above)
Comment (Bpendragon) [reply Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs]: fixed

**Review by eddynaka — APPROVED** [review]
:shipit:

**Pull request merged**